packages/common/src/multiple-choice-list.js

```diff
@@ -1,70 +1,52 @@
+const SingleCheckbox = (props) => {
+  const { input, Checkbox, ...rest } = useFieldApi({ ...props, type: 'checkbox' });
+
+  return <Checkbox {...input} {...rest} />;
+};
+
+const MultipleChoiceList = (props) => {
+  const { Wrapper, Checkbox, label, isRequired, helperText, meta, input, options, isDisabled, isReadOnly, description, ...rest } = useFieldApi(props);
+
+  const { error, touched } = meta;
+  const showError = touched && error;
 
   return (
-    <FieldProvider
-      {...props}
-      // eslint-disable-next-line no-undef
+    <Wrapper
+      showError={showError}
+      isRequired={isRequired}
+      label={label}
+      helperText={helperText}
+      meta={meta}
+      description={description}
+      rest={rest}
+      error={error}
       name={input.name}
-      validate={composeValidators(props.validate || [])}
-      render={({ label, isRequired, helperText, meta, options, isDisabled, isReadOnly, description, ...rest }) => {
-        const { error, touched } = meta;
-        const showError = touched && error;
-        const groupValues = rest.input.value || [];
-        return (
-          <Wrapper
-            showError={showError}
-            isRequired={isRequired}
-            label={label}
-            helperText={helperText}
-            meta={meta}
-            description={description}
-            rest={rest}
-            error={error}
-          >
-            {options.map((option) => (
-              <FieldProvider
-                formOptions={formOptions}
-                id={`${rest.id}-${option.value}`}
-                key={option.value}
-                {...option}
-                name={props.name}
-                type="checkbox"
-                render={({ input, meta, formOptions, componentType, ...rest }) => {
-                  const indexValue = groupValues.indexOf(input.value);
-                  return (
-                    <Checkbox
-                      aria-label={option['aria-label'] || option.label}
-                      {...input}
-                      {...rest}
-                      isDisabled={isDisabled || isReadOnly}
-                      onChange={() =>
-                        indexValue === -1
-                          ? input.onChange([...groupValues, input.value])
-                          : input.onChange([...groupValues.slice(0, indexValue), ...groupValues.slice(indexValue + 1)])
-                      }
-                      label={rest.label}
-                    />
-                  );
-                }}
-              />
-            ))}
-          </Wrapper>
-        );
-      }}
-    />
+    >
+      {options.map((option) => (
+        <SingleCheckbox
+          Checkbox={Checkbox}
+          aria-label={option['aria-label'] || option.label}
+          {...rest}
+          value={option.value}
+          label={option.label}
+          name={input.name}
+          option={option}
+          id={`${rest.id || input.name}-${option.value}`}
```

Comment:
The `rest.id` won't work on its own. All checkboxes will have the same id. Either add the -option.value or we should remove the id entirely and just use name/option combination.

Reply:
there is a `-${option.value}`